src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -347,6 +358,115 @@ def _retrieve_pre_processed_docs(tool: str,
     return docs
 
 
+def _convert_to_iso8601(date_string: str) -> str:
+    # Convert a date string to ISO 8601 format
+    formats_to_try = ["%Y/%m/%d", "%Y-%m-%d"]  # Add more formats as needed
+
+    for date_format in formats_to_try:
+        try:
+            parsed_date = datetime.strptime(date_string, date_format)
+            iso8601_date = parsed_date.date().isoformat()
+            return iso8601_date
+        except ValueError:
+            continue
+
+    raise ValueError("Unrecognized date format")
+
+
+def _convert_values_to_type(docs: list[dict], key: str, col_type: ColumnType):
+    # Convert the values of a column to the specified type
+    values = []
+    for doc in docs:
+        try:
+            value = doc[key]
+            value = None if value in NONE_STR else value
+            if value is None:
+                doc[key] = value
+                continue
+
+            if col_type == ColumnType.INT:
+                doc[key] = int(value)
+            elif col_type == ColumnType.FLOAT:
+                doc[key] = float(value)
+            elif col_type == ColumnType.STRING:
+                doc[key] = str(value)
+            elif col_type == ColumnType.DATE:
+                doc[key] = _convert_to_iso8601(value)
+            elif col_type == ColumnType.ENUM:
+                pass
```

Comment:
maybe we should just leave enums out for now so that when we add them an error occurs and we know to fix this code rather than having it silently ignore the enum

Reply:
👍

Reply:
:+1: